src/main/java/Duke.java

```diff
@@ -1,10 +1,147 @@
+                    int index = parser.getIndexToModify();
+                    Task task = taskList.getTask(index);
+                    taskList = taskList.deleteTask(index);
+                    ui.displayTaskDeleted(task, taskList);
+                    storage.writeFile(taskList);
+                    parser = parser.newInput(reader.nextLine());
+
+                    // user wants to add a ToDo
+                } else if (command.equals("todo")) {
+                    String taskDesc = parser.getTaskDescription();
+                    ToDo newTask = new ToDo(taskDesc);
+                    taskList = taskList.addTask(newTask);
+                    ui.displayTaskAdded(newTask, taskList);
+                    storage.writeFile(taskList);
+                    parser = parser.newInput(reader.nextLine());
+
+                    // user wants to add a Deadline
+                } else if (command.equals("deadline")) {
+                    String taskDesc = parser.getTaskDescription();
+                    LocalDate date = LocalDate.parse(parser.getDate(),
+                            DateTimeFormatter.ofPattern("d/MM/yyyy"));
+                    LocalTime time = LocalTime.parse(parser.getTime(),
+                            DateTimeFormatter.ofPattern("HHmm"));
+
+                    Deadline newTask = new Deadline(taskDesc, date, time);
+                    taskList = taskList.addTask(newTask);
+                    ui.displayTaskAdded(newTask, taskList);
+                    storage.writeFile(taskList);
+                    parser = parser.newInput(reader.nextLine());
+
+                    // user wants to add an Event
+                } else if (command.equals("event")) {
+                    String taskDesc = parser.getTaskDescription();
+                    LocalDate date = LocalDate.parse(parser.getDate(),
+                            DateTimeFormatter.ofPattern("d/MM/yyyy"));
+                    LocalTime time = LocalTime.parse(parser.getTime(),
+                            DateTimeFormatter.ofPattern("HHmm"));
+
+                    Event newTask = new Event(taskDesc, date, time);
+                    taskList = taskList.addTask(newTask);
+                    ui.displayTaskAdded(newTask, taskList);
+                    storage.writeFile(taskList);
+                    parser = parser.newInput(reader.nextLine());
+
+                // user wants to find tasks using a keyword
+                } else if (command.equals("find")) {
+                    String keyword = parser.getKeyword();
+                    ui.displayTaskSearch(keyword, taskList);
+                    parser = parser.newInput(reader.nextLine());
+
+                } else {
+                    throw new DukeException("unknown");
+                }
+            }
+
+        } catch (DukeException e) {
+            System.out.println(e.errorMessage());
+
+        } finally {
+            reader.close();
+        }
+    }
```

Comment:
Could this method be separated into multiple shorter methods?
https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods